catalyst/data/reader.py

```diff
@@ -117,11 +117,8 @@ def __call__(self, row):
             dtype: Scalar value
         """
         scalar = self.dtype(row.get(self.input_key, self.default_value))
-        if self.one_hot_classes is not None \
-                and scalar is not None and scalar >= 0:
-            one_hot = np.zeros(self.one_hot_classes, dtype=np.float32)
-            one_hot[scalar] = 1.0
-            scalar = one_hot
+        if self.one_hot_classes is not None:
+            scalar = get_one_hot(scalar, self.one_hot_classes)
```

Comment:
could you please propagate `smoothing` too?

Reply:
okay

Reply:
done